File: Cpp/941_Valid_Mountain_Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& A) {
        if(A.size() <3) return false;
        bool ascending = true;
        for(int i=1; i<A.size(); ++i){
            if(A[i] > A[i-1]) {
                if(!ascending) return false;
            } else if(A[i] < A[i-1]) {
                if(i==1) return false;
                ascending = false;
            } else {
                return false;
            }
        }
        
        return !ascending;
    }
};

File: Kotlin/941_Valid_Mountain_Array.kt
class Solution {
    fun validMountainArray(arr: IntArray): Boolean {
       	if(arr.size < 3) return false
        
        var increasing = true
        for(i in 1 until arr.size){
            if(arr[i] > arr[i-1]){
                if(!increasing) return false
            }else if(arr[i] < arr[i-1]){
                if(i == 1) return false
                increasing = false
            }else{
                return false
            }
        }
        
        return !increasing
    }
}